scripts/beam-sql.sh

```diff
@@ -0,0 +1,260 @@
+  echo "distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.9.6/apache-maven-3.9.6-bin.zip" > "${wrapper_dir}/maven-wrapper.properties"
+
+  # Download the mvnw script and the wrapper JAR
+  curl -sSL -o "${WORK_DIR}/mvnw" "${mvnw_script_url}"
+  curl -sSL -o "${wrapper_dir}/maven-wrapper.jar" "${wrapper_jar_url}"
+
+  # Make the wrapper script executable
+  chmod +x "${WORK_DIR}/mvnw"
+}
+
+function usage() {
+  echo "Usage: $0 [--version <beam_version>] [--runner <runner_name>] [--io <io_connector>]..."
+  echo ""
+  echo "A self-contained launcher for the Apache Beam SQL Shell."
+  echo ""
+  echo "Options:"
+  echo "  --version   Specify the Apache Beam version (default: ${DEFAULT_BEAM_VERSION})."
+  echo "  --runner    Specify the Beam runner to use (default: direct). Supported: direct, dataflow."
+  echo "  --io        Specify an IO connector to include (e.g., iceberg, kafka). Can be used multiple times."
+  echo "  --list-versions      List all available Beam versions from Maven Central and exit."
+  echo "  -h, --help  Show this help message."
+  exit 1
+}
+
+# This function fetches all available Beam versions from Maven Central.
+function list_versions() {
+  echo "🔎 Fetching the 10 most recent Apache Beam versions from Maven Central..."
+  local metadata_url="https://repo1.maven.org/maven2/org/apache/beam/beam-sdks-java-core/maven-metadata.xml"
+
+  if ! command -v curl &> /dev/null; then
+    echo "❌ Error: 'curl' is required to fetch the version list." >&2
+    return 1
+  fi
+
+  # Fetch, parse, filter, sort, and take the top 10.
+  local versions
+  versions=$(curl -sS "${metadata_url}" | \
+    grep '<version>' | \
+    sed 's/.*<version>\(.*\)<\/version>.*/\1/' | \
+    grep -v 'SNAPSHOT' | \
+    sort -rV | \
+    head -n 10) # Limit to the first 10 lines
+
+  if [ -z "${versions}" ]; then
+    echo "❌ Could not retrieve versions. Please check your internet connection or the Maven Central status." >&2
+    return 1
+  fi
+
+  echo "✅ 10 latest versions:"
+  echo "${versions}"
+}
+
+# This function ensures our temporary directory is cleaned up when the script exits.
+function cleanup() {
+  rm -rf "${WORK_DIR}"
+}
+trap cleanup EXIT # Register the cleanup function to run on script exit
+
+# --- Argument Parsing ---
+BEAM_VERSION="${DEFAULT_BEAM_VERSION}"
+IO_CONNECTORS=()
+BEAM_RUNNER="direct"
+SQLLINE_ARGS=()
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --version) BEAM_VERSION="$2"; shift ;;
+    --runner) BEAM_RUNNER=$(echo "$2" | tr '[:upper:]' '[:lower:]'); shift ;;
+    --io) IO_CONNECTORS+=("$2"); shift ;;
+    --list-versions) list_versions; exit 0 ;;
+    -h|--help) usage ;;
+    *) SQLLINE_ARGS+=("$1") ;;
+  esac
+  shift
+done
+
+# --- Prerequisite Check ---
+# Java is always required.
+if ! command -v java &> /dev/null; then
+  echo "❌ Error: 'java' command not found. It is required to run the application." >&2
+  exit 1
+fi
+
+# Decide which Maven command to use. Prefer system 'mvn'.
```

Comment:
Prefer maven wrapper would make the script more reliable. System mvn might be a random version and end up incompatible with something, if the generated pom gets more complex. (though probably it will always be trivial and if it gets complex we will switch to a different way of resolving deps)

Reply:
Yeah you are right. I made change accordingly